services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestsHandler.java

```diff
@@ -467,13 +472,17 @@ private CompletableFuture<ReadResponse> handleMultiGetRequestInParallel(
             try {
               responseWrapper.addRecord(record);
 
-              if (optionalValueList.isPresent()) {
-                optionalValueList.get().add(record.value.remaining());
+              if (responseValueSizeList != null) {
+                responseValueSizeList.set(subChunkCur, record.value.remaining());
               }
```

Comment:
This can be moved out of the lock since we are setting by index.

Reply:
Sure